libs/prisma-models/src/pk.rs

```diff
@@ -0,0 +1,38 @@
+use std::sync::Arc;
+
+use crate::{Field, ScalarFieldRef, ScalarFieldWeak};
+
+#[derive(Debug)]
+pub struct PrimaryKeyTemplate {
```

Comment:
I don't believe that you really need the template pattern here - we only use it for the other things because of cycles, but since you can construct the PK from the finalized model template, I don't think you need this.